src/braket/quantum_information/pauli_string.py

```diff
@@ -142,6 +147,186 @@ def eigenstate(self, signs: Optional[Union[str, List[int], Tuple[int, ...]]] = N
+            elif self._nontrivial[i] == other._nontrivial[i]:
+                pauli_result += "I"
+            else:
+                gate, phase = _PRODUCT_MAP[self._nontrivial[i]][other._nontrivial[i]]
+                pauli_result += gate
+                phase_result *= phase
+
+        # ignore complex global phase
+        if phase_result.real < 0 or phase_result.imag < 0:
+            pauli_result = "-" + pauli_result
+        out_pauli_string = PauliString(pauli_result)
+
+        if inplace:
+            self._phase = out_pauli_string._phase
+            self._qubit_count = out_pauli_string._qubit_count
+            self._nontrivial = out_pauli_string._nontrivial
+        return out_pauli_string
+
+    def __mul__(self, other: PauliString) -> PauliString:
+        """Right multiplication operator overload using `dot()`.
+
+        Returns the result of multiplying the current circuit by the argument on its right.
+
+        Args:
+            other (PauliString): The right multiplicand.
+
+        Returns:
+            PauliString: The resultant circuit from right multiplying `self` with `other`.
+
+        Raises:
+            ValueError: If the lengths of the Pauli strings being multiplied differ.
+
+        See Also:
+            `braket.quantum_information.PauliString.dot()`
+        """
+        return self.dot(other)
+
+    def __imul__(self, other: PauliString) -> PauliString:
+        """Operator overload for right-multiplication assignment (`*=`) using `dot()`.
+
+        Right-multiplies `self` by `other`, and assigns the result to `self`.
+
+        Args:
+            other (PauliString): The right multiplicand.
+
+        Returns:
+            PauliString: The resultant circuit from right multiplying `self` with `other`.
+
+        Raises:
+            ValueError: If the lengths of the Pauli strings being multiplied differ.
+
+        See Also:
+            `braket.quantum_information.PauliString.dot()`
+        """
+        return self.dot(other, inplace=True)
+
+    def power(self, n: int, inplace: bool = False) -> PauliString:
+        """Composes Pauli string with itself n times.
+
+        Args:
+            n (int): The number of times to self-multiply. Can be any integer value.
+            inplace (bool): Update `self` if `True`
+
+        Returns:
+            PauliString: If `n` is positive, result from self-multiplication `n` times.
+            If zero, identity. If negative, self-multiplication from trivial
+            inverse (recall Pauli operators are involutory).
+
+        Raises:
+            ValueError: If `n` isn't a plain Python `int`.
+        """
+        if not isinstance(n, int):
+            raise ValueError("Must be raised to integer power")
+
+        pauli_other = PauliString(self)
+        if n == 0:
+            pauli_other._phase = 1
+            pauli_other._nontrivial = {}
+        else:
+            for _ in range(n - 1):
+                pauli_other.dot(self, inplace=True)
```

Comment:
Try to take advantage of the fact that the Pauli matrices are involutory; $\sigma_i^k = \sigma_i^{k \\, mod \\, 2}$.